internal/controller/database_controller.go

```diff
@@ -388,7 +396,12 @@ func (r *DatabaseReconciler) deleteDatabase(ctx context.Context, dbcr *kindav1be
 		Username: dbcr.Status.UserName,
 	}
 
-	db, dbuser, err := determinDatabaseType(dbcr, databaseCred)
+	instance := &kindav1beta1.DbInstance{}
+	if err := r.Get(ctx, types.NamespacedName{Name: dbcr.Spec.Instance}, instance); err != nil {
```

Comment:
(minor)

had some head scratching on this line. Hard to understand.

Reply:
@dabde 

You mean this one?
```go
if err := r.Get(ctx, types.NamespacedName{Name: dbcr.Spec.Instance}, instance); err != nil {
```

Would it better, if I rewrite it like that

```
nsname := types.NamespacesName{Name: dbcr..Spec.Instance}
err = r.Get(ctx, nsname, instance)
if err != nil {
```

Reply:
No, more that you need to know that `r.Get(x,x, pointer)` interface of the function.